### PR TITLE
Group 10 python2 to 3 conversion

### LINKED ISSUE #19747: Convert group 10 from python2 to python 3

### PR DESCRIPTION
**Pull request by granrothge**
Updating code to be python 3 compliant as part of umbrella ticket #18550.

**To test:**

Code inspection and passing tests should be enough.

Fixes #19747

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def runTest(self):
 
         # Save the masked spectra nmubers to a simple ASCII file for comparison
         self.saved_diag_file = os.path.join(ms.config['defaultsave.directory'], 'CurrentDirectInelasticDiag.txt')
-        handle = file(self.saved_diag_file, 'w')
+        handle = open(self.saved_diag_file, 'w')
```
Comment (peterfpeterson) [inline Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic.py]: Instead this should be
```python
with open(self.saved_diag_file, 'w') as handle:
```
then indent the code below and remove the explicit `close`

File: Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic2.py [hunk 2]
```diff
@@ -1,5 +1,11 @@
+from __future__ import (absolute_import, division, print_function)
 #pylint: disable=invalid-name,no-init
 import os
+#python 2 to 3 changes
```
Comment (peterfpeterson) [inline Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic2.py]: remove these comments

File: Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic2.py [hunk 3]
```diff
@@ -70,7 +76,7 @@ def runTest(self):
         # Save the masked spectra numbers to a simple ASCII file for comparison
         self.saved_diag_file = os.path.join(config['defaultsave.directory'],
                                             'CurrentDirectInelasticDiag2.txt')
-        handle = file(self.saved_diag_file, 'w')
+        handle = open(self.saved_diag_file, 'w')
```
Comment (peterfpeterson) [inline Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic2.py]: Same comment as `Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic.py`

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

File: Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic2.py (hunk 3 above)
Comment (granrothge) [inline Testing/SystemTests/tests/analysis/DirectInelasticDiagnostic2.py]: Updates have been made as requested.

**Review by peterfpeterson — APPROVED** [review]
Looks good once the builds pass

**Comment by martyngigg** [comment]
Sorry for stepping in here. I resolved what I thought was a trivial merge conflict and then `flake8` moaned again on a different file.

**Pull request merged**